tools/omnissm/pkg/aws/ssm/utils.go

```diff
@@ -14,10 +14,25 @@
 
 package ssm
 
-import "strings"
+import (
+	"regexp"
+	"strings"
+)
 
 // IsManagedInstance tests whether the provided EC2 instance identifier is
 // managed.
 func IsManagedInstance(s string) bool {
 	return strings.HasPrefix(s, "mi-")
 }
+
+// original Java regexp for a valid tag: ^([\p{L}\p{Z}\p{N}_.:/=+\-@]*)$
+var invalidTagRegexp = regexp.MustCompile(`[^a-zA-Z0-9\s_.:/=+\-@]`)
+
+func SanitizeTag(t string) string {
+	// special case of key: value# comment is this
+	if i := strings.Index(t, "#"); i != -1 {
```

Comment:
could you clarify the case trying to be addressed here, is it trailing '#' or any '#' if its any why not part of invalid regex substitution, if its the former, given a string `val#ue` -> `val`

Reply:
Good question - this was meant to specifically address the case I've seen most often, where a yaml comment isn't separated from the tag value by a space (i.e.:

```yaml
...
Tags:
  App: MYAPP# The app name
```
Which comes through as `{"App": "MYAPP# The app name"}`.

In this case the intent was to try to salvage a (potentially) valid tag value, knowing that this was a common case. The following regex should strip out any remaining invalid characters to avoid causing an SSM error otherwise.